api/v1alpha1/bucketcontent_types.go

```diff
@@ -17,25 +17,96 @@ limitations under the License.
 package v1alpha1
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+// Regions where AWS S3 API is available
+type S3Region string
+
+const (
+	// S3RegionUSEast1 = us-east-1
+	S3RegionUSEast1 S3Region = `json:"us-east-1"`
+)
+
+// S3ProtocolSpec is the specification for S3 and S3 compatible Object Storage services
+type S3ProtocolSpec struct {
+	// Endpoint is the URL for accessing the S3 endpoint
+	// +optional
+	Endpoint string `json:"endpoint,omitempty"`
+
+	// BucketName is the name of the S3 or S3 compatible bucket
+	BucketName string `json:"bucket"`
+
+	// Region is the Region where the bucket is situated
+	// +optional
+	Region S3Region `json:"region,omitempty"`
+
+	// SignatureVersion is the version of the S3 API signature
+	// +optional
+	SignatureVersion string `json:"signatureVersion,omitempty"`
+
+	// ServiceAccount is the service account for the BucketContent
+	// +optional
+	ServiceAccount string `json:"serviceAccount,omitempty"`
+}
+
+// ProtocolSpec defines the specification of the Object Storage protocol
+type ProtocolSpec struct {
+	// ProtocolType is the type of the underlying protocol for the associated Bucket
+	ProtocolType Protocol `json:"type"`
+
+	// S3ProtocolSpec is the specification for S3 and S3 compatible Object Storage services
+	// +optional
+	S3ProtocolSpec S3ProtocolSpec `json:"s3,omitempty"`
+
+	// AzureProtocolSpec
+	// GCSProtocolSpec
+}
+
+// ReleasePolicy defines the lifecycle of the bucket once the pod(s) using it have exited
+type ReleasePolicy string
+
+const (
+	// ReleasePolicyRetain denotes that the bucket must be retained after it is used
+	ReleasePolicyRetain ReleasePolicy = "retain"
+
+	// ReleasePolicyDelete denotes that the bucket must be deleted after it is used
+	ReleasePolicyDelete ReleasePolicy = "delete"
+)
 
 // BucketContentSpec defines the desired state of BucketContent
 type BucketContentSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// Provisioner is the name of the provisioner as defined in the BucketClass of this BucketContent
+	Provisioner string `json:"provisioner"`
+
+	// ReleasePolicy defines the lifecycle of the bucket once the pod(s) using it have exited
+	// +optional
+	ReleasePolicy ReleasePolicy `json:"releasePolicy,omitempty"`
+
+	// AccessMode defines the restriction of access allowed for this BucketContent
+	// +optional
+	AccessMode string `json:"accessMode,omitempty"`
```

Comment:
Thinking on the name - AccessMode is somewhat specific to file/block and isn't really descriptive of what it represents, which is public access to the the bucket.  What do you think about instead calling it `AnonymousAccessMode`.  This I think conveys the ACLs of the bucket itself, and how it can be accessed publicly, without a user identity.

Reply:
Does it need to be a string? Is enumerated or bool better?
+1 on `anonymous` in the name.